packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -133,16 +130,28 @@ export class DataProxyEngine extends Engine {
     }
   }
 
-  request<T>({ query }: EngineQuery, { headers = {}, transaction }: RequestOptions<DataProxyTxInfoPayload>) {
+  request<T>(
+    { query }: EngineQuery,
+    { traceparent, transaction, customDataProxyHeaders }: RequestOptions<DataProxyTxInfoPayload>,
+  ) {
     this.logEmitter.emit('query', { query })
 
+    const headers: Record<string, string> = { ...customDataProxyHeaders }
+    if (traceparent) {
+      headers.traceparent = traceparent
+    }
+
+    if (transaction) {
+      headers['X-transaction-id'] = transaction.id
```

Comment:
I think this can possibly send a batch id, is it intentional?

Reply:
No, singular request receives only iTx options

Reply:
Also, batch ids are not propagated to `engine-core` - `transaction: { kind: 'batch}` has no id field.